packages/flutter/lib/src/painting/text_painter.dart

```diff
@@ -217,6 +217,86 @@ class TextPainter {
        _textWidthBasis = textWidthBasis,
        _textHeightBehavior = textHeightBehavior;
 
+  /// Computes the width of a configured [TextPainter].
+  ///
+  /// This is a convenience method that creates a text painter with the supplied
+  /// parameters, lays it out with the supplied [minWidth] and [maxWidth], and
+  /// returns its [TextPainter.width] making sure to dispose the underlying
+  /// resources.
+  static double computeWidth({
+    required InlineSpan text,
+    TextAlign textAlign = TextAlign.start,
+    TextDirection? textDirection,
```

Comment:
TextDirection must be non null I think?

Reply:
This was copied from the `TextPainter` ctor. I'd like to keep the types the same for ease of migration.

If we want to update this we should update the ctor as well.

Reply:
Ahh but I see what you're saying, on the regular object we can set this to some value later, whereas we can't with this method. Ok. In that case, I'll file a follow upt o make it non-nullable.